src/codegen/mod.rs

```diff
@@ -3948,11 +3959,45 @@ pub(crate) fn codegen(
             &(),
         );
 
+        // If the set of items to generate dynamic bindings for is nonempty...
+        if !context.dyngen_items().is_empty() {
```

Comment:
So basically... This is just effectively generating all the functions right now, is that right?

The way I'd have approached this would be something like, maybe:

 * Add an extra field to `CodegenResult` (`dynamic_functions`? / whatever)
 * Handle this in `CodeGenerator for Function` (just something like `if ctx.options().dynamic_loading { /* generate appropriate things */ }`, and instead of pushing to the regular `CodegenResult` field, push it to the dynamic functions, which you'd then handle here.

Is there any reason such an approach wouldn't work? It'd handle a lot more edge cases than this (template functions, etc, which this patch doesn't handle), and should also be much simpler / not need extra `ItemSet`s.